### ISSUE TITLE
Consider dropping most environment variables

### ISSUE THREAD

**Issue by jamessynge** [panoptes/POCS#555]
We have PANDIR, POCS, etc. And then we have config files that have the values of those envvars. It seems like we should have as few envvars as possible, and infer them where possible, or record them in config files where not.

@wtgee and @AnthonyHorton, your thoughts please.

**Comment by jamessynge** [comment]
I was reviewing @AnthonyHorton's change which adds a pyro config file, and that includes a path starting with /var/pantoptes. It seems to me that we should try to avoid duplicating the paths.

So, I'd like to either extend our YAML support with the ability to evaluate env vars, or reduce the use of the env vars... or both if there are some that really aren't needed.

**Comment by wtgee** [comment]
A mapping for directories specifically? Might be a good idea to pull that out of pocs.yaml and put directly in $PANDIR.

**Comment by AnthonyHorton** [comment]
I'm supportive of the idea of removing the dependence on environment variables, and instead putting the items we really need in the config files and inferring the rest. On general principles keeping all the important configuration in the `conf_files` directory, rather than having some of it in user's `.bashrc`, `.profile`, etc. seems like a good idea to me.

FWIW the setting of `base: /var/panoptes` in the `pyro_camera.yaml` file that I added isn't a new thing. The `directories` section of that file is just a copy of the `directories` section in `pocs.yaml`. We've had a situation for a long time were we have an environment variable for something (`PANDIR`) as well as a config entry for the same thing (`config['directories']['base']`), which is obviously a recipe for confusing errors if they ever have different values. We should definitely be setting these things in only one place, & my preference would be the config files.

Having `$POCS`, `$PANDIR`, etc. defined in your shell can be handy, but of course it will still be possible to do that even if POCS itself no longer uses the environment variables. We could make that easy by having a script that parses the config files and sets the corresponding environment variables for you, or adds them to your `.profile`.

In terms of what to include `$PANDIR` and `$PANUSER` (or their config file equivalents) seem to be the only essential ones.

**Comment by wtgee** [comment]
Huntsman runs everything as `huntsman` user under `/var/huntsman`. Works great to just set PANUSER and PANDIR. I believe our travis testing also uses PANUSER.

I've started stripping a few env variables in the PRs I am working on.

I've added #564 with some other config related items I've been thinking about. Please add anything to that overview issue.

**Comment by wtgee** [comment]
The `docker` branch (soon to be merged into `develop` via #951) assumes there is an environment file in `$PANDIR/.env` (can be read via [dotenv](https://pypi.org/project/python-dotenv/)). This environment file currently consists of the following:

```bash
PANDIR=/var/panoptes
POCS=/var/panoptes/POCS
PANLOG=/var/panoptes/logs
```

Additionally, for those running the docker services there should be a:

```
LOCAL_USER_ID=1000
```

(This may disappear in future but is necessary now. Also technically it should be the output from `id -u` not just `1000`).

If accessing [GCP](https://cloud.google.com) services then there should also be a:

```bash
GOOGLE_APPLICATION_CREDENTIALS=/path/to/service-account-key.json
```

The `.env` file helps start up the config server and everything else should be queried via the config server and not rely on env vars.

This means that there must be a one-time manual configuration of the `.env` file (i.e. it can't be completely self-discoverable).

I'm not sure there is much utility in keeping this broad issue open.  As part of my issue cleanup I'm closing this. Can be reopened if needed.
